msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -2291,5 +2394,33 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
         val result = application.signUp(invalidUsername)
         assertTrue(result is SignUpError)
         assertTrue((result as SignUpError).isInvalidUsername())
+
+        checkSafeLogging()
+    }
+
+    private fun checkSafeLogging() {
+        val piiTrueToCheck = listOf(
+            "password:","password=",
+            "code:","code=",
+            "continuationToken:","continuationToken="
+        )
+        val piiFalseToCheck = listOf(
+            "username:", "username="
+        )
+
+        val elementsToCheck = piiTrueToCheck.toMutableList()
+
+        if (!allowPII) {
+            elementsToCheck.addAll(piiFalseToCheck)
+        }
+
+        elementsToCheck.forEach { element ->
+            verifyLogDoesNotContain("Command", element)
+        }
+
+        Logger.getInstance().setExternalLogger(null)
+    }
+    private fun verifyLogDoesNotContain(tag: String, element: String) {
+        verify(externalLogger, never()).log(contains(tag), any(), contains(element), eq(true))
```

Comment:
We're actually doing many tests in 1 statement here. While it's efficient from a coding perspective, it may not be logistically. if this test fails, it's hard to determine why, because so many things happen in the same statement.

Consider something like:
- attach a real logger callback interface (make it a Spy, so the method is actually called, but you can also play around with its behaviour the way you can with a Mock)
- within the callback, perform a number of assertions all on different fields and callback configurations, each in a separate statement.

Reply:
Updated according to the suggestions.